packages/reactv2/src/hooks/useFlow.ts

```diff
@@ -1,29 +1,35 @@
-import { Frigade, Flow } from '@frigade/js'
-import { useContext, useState } from 'react'
+import { Flow, Frigade } from '@frigade/js'
+import { useContext, useEffect, useState } from 'react'
 
 import { FrigadeContext } from '../components/Provider'
 
 export function useFlow(flowId: string) {
   const [flow, setFlow] = useState<Flow>(null)
   const { apiKey, config } = useContext(FrigadeContext)
+  const filteredConfig = Object.fromEntries(
+    Object.entries(config).filter(([k, v]) => ['apiUrl', 'userId'].includes(k) && v != null)
+  )
+  const frigade = new Frigade(apiKey, filteredConfig)
+
+  useEffect(() => {
+    frigade.onFlowStateChange((flow) => {
+      setFlow(flow)
+    })
+    return () => {
+      frigade.removeOnFlowStateChangeHandler((flow) => {
```

Comment:
I don't think this will remove the handler, since it's a new inline function and not a reference to the original on line 15.

Reply:
Ah whoops! Good call.